contents/data/sql/SQL.md

```diff
@@ -0,0 +1,154 @@
+Rather, RDBMSs %%like MySQL, Oracle, SQL Server and PostgreSQL%% come with an _implementation_ of SQL. Note that the syntaxes may <tooltip content="in terms of whether the syntax is case-sensitive, the format for specifying date and time, what functions are available out-of-the-box">vary</tooltip> across these RDBMSs.
+
+These RDBMSs come with a GUI application, but can be accessed from the command line as well. Figure 1 shows MySQL Workbench GUI software and MySQL in the command line.
+
+<img src="MySQL_Workbench.png" alt="MySQL Workbench GUI Software" width="57%">
+<img src="MySQL_cli.png" alt="Running MySQL from the command line" width="40%">
+
+_Figure 1. MySQL can be accessed using the GUI or command line_ <sup>[source 1](https://www.mysql.com/products/workbench/)</sup> <sup>[source 2](https://www.researchgate.net/figure/The-MySQL-command-line-tool_fig5_328093393)</sup>
+
+---
+
+### How does SQL work?
+
+SQL uses **queries** to retrieve data. Here is an example of how an SQL query is used.
+
+Suppose we have the data table `Students` shown here:
+
+| ID | Name  | Course | Faculty |
+| -- | ----- | ------ | ------- |
+| 1  | Alex  | CS202  | CS      |
+| 2  | Bob   | MA303  | MA      |
+| 3  | Cathy | CS202  | CS      |
+| 4  | Daren | CS202  | CS      |
+| 5  | Ellie | CS101  | CS      |
+| 6  | Fred  | MA303  | MA      |
+| 7  | Gary  | CS101  | CS      |
+| 8  | Henry | CS404  | CS      |
+
+We can use this SQL query to retrieve information from this table:
+```sql
+SELECT Course, COUNT(*) num
+FROM Students
+WHERE Faculty = 'CS'
+GROUP BY Course
+HAVING COUNT(*) > 1
+ORDER BY num;
+```
+
+This query first filters the entries in the `Students` table such that only entries that have `CS` as the faculty are considered.
+After that, it groups those entries into the 3 courses: `CS101`, `CS202` and `CS404`.
+Then, it removes courses that do not have more than 1 student, i.e., `CS404` is removed from consideration.
+Finally, it returns a list of courses with a count of the number of students, ordered in increasing order.
+So, the output of the query is:
+
+| Course | num |
+| ------ | --- |
+| CS202  | 3   |
+| CS101  | 2   |
+
+<br/>
+You can see how this simple query can prove to be extremely useful for getting this information when the table has tens of entries (or many more). Apart from retrieving information, SQL can also be used for creating, deleting and manipulating data with queries like `INSERT`, `DELETE` and `UPDATE` for entries, in addition to `CREATE`, `DROP` and `ALTER` for tables as a whole.
+
+<br/>
+For example, you can set up the structure of table shown in Table 1 by executing the `CREATE TABLE` query shown here.
+It creates a new data table `Students`, with 4 fields (`ID`, `Name`, `Course` and `Faculty`) and specifies their data types.
```

Comment:
No need for `<br/>`? Anyway, you need a blank line before markdown content. Without it, this paragraph doesn't render as expected.
![image](https://user-images.githubusercontent.com/1673303/55242894-219d7b00-5279-11e9-8897-ea62965a619e.png)

Reply:
Ooh... I was wondering why that wasn't rendering properly. Thanks!